src/systemTest/java/com/flightstats/hub/system/service/ChannelService.java

```diff
@@ -71,6 +85,77 @@ public String addItem(String channelName, Object data) {
         return response.body().get_links().getSelf().getHref();
     }
 
+    private List<String> getPathParts(String path) {
+        String itemPath = path.replace(hubBaseUrl.toString() + "channel/", "");
+        return Arrays.asList(itemPath.split("/"));
+    }
+
+    private Map<DatePathIndex, Integer> getPathDateParts(List<String> pathParts) {
+        Map<DatePathIndex, Integer> dateValues = new HashMap<>();
+        List<Integer> dateParts = pathParts.subList(1, pathParts.size() -1).stream()
+                .filter(StringUtils::isNotBlank)
+                .map(Integer::parseInt)
+                .collect(Collectors.toList());
+        dateValues.put(DatePathIndex.YEAR, dateParts.get(DatePathIndex.YEAR.getIndex()));
```

Comment:
Might be nice to push the logic into DatePathIndex, so the code would look like:
```
dateValues.put(DatePathIndex.YEAR.extract(dateParts));
dateValues.put(DatePathIndex.MONTH.extract(dateParts));
etc
etc
```
Or even have a method on DatePathIndex that knows how to take a List<Integer> and return a Map:
```
dateValues = DateValueIndex.extractDates(dateParts);
```